apps/src/lab2/levelEditors/aichatSettings/CollapsibleFieldSection.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+import {AiCustomizations} from '@cdo/apps/aichat/types';
+import CollapsibleSection from '@cdo/apps/templates/CollapsibleSection';
+
+import FieldSection from './FieldSection';
+
+import moduleStyles from './edit-aichat-settings.module.scss';
+
+interface CollapsibleFieldSectionProps {
+  fieldName: keyof AiCustomizations;
+  labelText: string;
+  description?: string;
+  inputType: 'text' | 'number' | 'textarea' | 'custom';
+  inputNode?: React.ReactNode;
+  min?: number;
+  max?: number;
+  step?: number;
+}
+
+const CollapsibleFieldSection: React.FunctionComponent<
+  CollapsibleFieldSectionProps
+> = ({
+  fieldName,
+  description,
+  labelText,
+  inputType,
+  inputNode,
+  min,
+  max,
+  step,
```

Comment:
Similarly, might be able to do something like
```
const CollapsibleFieldSection: React ... = ({ labelText, ...rest }) => {
   ...
   <CollapsibleSection headerContent={labelText}>
     <FieldSection {...rest } />
```
Not 100% that TS won't complain but I do think there's something we can do here to minimize redundancy

Reply:
Slick!